contracts/root/withdrawManager/WithdrawManagerStorage.sol

```diff
@@ -4,15 +4,24 @@ import { ProxyStorage } from "../../common/misc/ProxyStorage.sol";
 import { Registry } from "../../common/Registry.sol";
 import { RootChain } from "../RootChain.sol";
 
+contract ExitsDataStructure {
+  struct Input {
+    address signer;
+  }
 
-contract WithdrawManagerHeader {
   struct PlasmaExit {
     address owner;
     address token;
     uint256 receiptAmountOrNFTId;
+    bytes32 txHash;
     bool burnt;
+    address predicate;
```

Comment:
instead of having predicate and token both field we can choose one of, and have a `mapping` to reduce onchain storage.
is `predicate <=> token` relation [1:1,1:m,m:1,m:m] where m=many
other then m:m we can have mapping outside and save data.

Reply:
I think this would be many-to-many since for eg. an `erc20` predicate and a marketplace predicate both could start an erc20 exit.

Reply:
ah! M:M will add a lot of cost to user exits.

Reply:
How so? They will call `startExit` in the predicate of their choice. What am I missing?

Reply:
I was just thinking in terms of 1 extra storage cost per exit.